examples/video-chat/screens/PreviewScreen/index.tsx

```diff
@@ -78,12 +78,14 @@ const PreviewScreen = ({ navigation, route }: Props) => {
   };
 
   useEffect(() => {
-    getCaptureDevices().then((devices) => {
+    async function getCurrentCamera() {
+      const devices = await getCaptureDevices();
       availableCameras.current = devices;
       setCurrentCamera(devices.find((device) => device.isFrontFacing) || null);
-    });
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+    }
+
+    getCurrentCamera();
+  }, [getCaptureDevices]);
 
```

Comment:
We could memo `availableCameras`. But its is not necessary.

Reply:
If I understand correctly, you want to call `getCaptureDevices` inside `useMemo`. I don't know if this would be helpful TBH. `availableCameras` is variable that do not influence object state, so it can be cached in static way (with `useRef`).

Reply:
let it be.